### PR TITLE
Jetpack Forms: Change the message sent Tracks event to a bump stat

### PR DESCRIPTION
**Pull request by donnchawp**
The message sent event is a Tracks event but it really should be a bump stat.

This PR converts the contact_form_block_message_sent Tracks event to a bump stat called "jetpack_forms_message_sent". The extra value of the stat is the publish week of the post containing the form.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
panfyZ-1VS-p2#comment-4776

## Does this pull request change what data or activity we track or use?
Yes, it reduces the amount of data we track.

## Testing instructions:
Apply patch
It might be easier to test if you add an error_log() to jetpack_tracks_record_grunion_pre_message_sent() so you see the code is reached.
Create a form on a post.
Fill in the form and submit it.
Check your php error_log.
Automatticians can check the bump stat for jetpack_forms_message_sent.

### CODE REVIEW

File: projects/packages/forms/src/contact-form/class-util.php [hunk 1]
```diff
@@ -262,68 +262,19 @@ public static function grunion_delete_old_spam() {
...
-				$master_user_id = \Jetpack_Options::get_option( 'master_user' );
-				if ( ! empty( $master_user_id ) ) {
-					$event_user = get_userdata( $master_user_id );
-				}
-			}
-
-			$tracking = new \Automattic\Jetpack\Tracking();
-			$tracking->record_user_event( $event_name, $event_props, $event_user );
-		}
+		do_action( 'jetpack_bump_stats_extras', 'jetpack_forms_message_sent', $extra );
```
Comment (jeherve) [inline projects/packages/forms/src/contact-form/class-util.php:278]: Let's add a comment to satisfy the codex builder:

```suggestion
		/** This action is documented in jetpack/modules/widgets/social-media-icons.php */
		do_action( 'jetpack_bump_stats_extras', 'jetpack_forms_message_sent', $extra );
```

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php [hunk 2]
```diff
@@ -5209,67 +5209,18 @@ function grunion_delete_old_spam() {
...
-			$master_user_id = Jetpack_Options::get_option( 'master_user' );
-			if ( ! empty( $master_user_id ) ) {
-				$event_user = get_userdata( $master_user_id );
-			}
-		}
-
-		$tracking = new Automattic\Jetpack\Tracking();
-		$tracking->record_user_event( $event_name, $event_props, $event_user );
-	}
+	do_action( 'jetpack_bump_stats_extras', 'jetpack_forms_message_sent', $extra );
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php:5336]:
```suggestion
	/** This action is documented in modules/widgets/social-media-icons.php */
	do_action( 'jetpack_bump_stats_extras', 'jetpack_forms_message_sent', $extra );
```

**Review by jeherve — DISMISSED** [review]

**Review by jeherve — COMMENTED** [review]
This makes sense, but would this work on self-hosted and WoA sites? We do not have anything hooked into `jetpack_bump_stats_extras` on self-hosted sites. We do on WordPress.com so I assume this will work there, ~although the matching MC page doesn't exist yet, even though I made a few tests on WordPress.com and on self-hosted, and I assume you had made your own tests earlier?~ Scratch that, my test on a WordPress.com site came through, the page now exists.

Since the stats are already stored per day, do we need to track the year-week in this scenario?

**Comment by donnchawp** [comment]
We still need the year-month extra value for segmenting data.

**Comment by jeherve** [comment]
For more discussion on this, see p1678723950814709-slack-C01CSBEN0QZ

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
Approving this following discussion in Slack.

This will not work on WoA or self-hosted sites, but that's acceptable for now:

p1678819264711989/1678723950.814709-slack-C01CSBEN0QZ

**Pull request merged**